specs/data_structures.md

```diff
@@ -920,6 +923,24 @@ For explanation on entries, see the [reward distribution rationale document](../
 
 Represents a (potentially) non-integer number.
 
+### MessagePaid
+
+| name     | type                      | description                                                           |
+| -------- | ------------------------- | --------------------------------------------------------------------- |
+| `start`  | `uint64`                  | Share index (in row-major order) of first share paid for (inclusive). |
+| `finish` | `uint64`                  | Share index (in row-major order) of last share paid for (inclusive).  |
```

Comment:
Nit: of first/last share paid for in this block? Or by a given set of Txs?

Reply:
Neither? For the transaction with the ID that's the same as the key of this state element.

Reply:
I see.